runtime/manifest.js

```diff
@@ -219,6 +230,40 @@ class Manifest {
           connection.connectToView(targetView);
         }
       }
+      for (let slotConnectionItem of item.slotConnections) {
+        let slotConn = particle.addSlotConnection(slotConnectionItem.param, slotConnectionItem.dir);
+        if (!slotConnectionItem.name) {
+          continue;
+        }
+        assert(items.byName.has(slotConnectionItem.name));
+        let {item, slot} = items.byName.get(slotConnectionItem.name);
+        slotConn.connectToSlot(slot);
+        if (item.ref.view || (item.ref.tags.length > 0)) {
+          let foundConnection = false;
+          for (let particleViewConn of Object.values(particle.connections)) {
+            if (!particleViewConn.view) continue;
+            if ((particleViewConn.view.localName == item.ref.view) ||
+                (item.ref.tags.every(t => particleViewConn.view.tags.indexOf(t) >= 0))) {
+              slotConn.connectToView(particleViewConn.name);
+              foundConnection = true;
+              break;
+            }
+          }
+          // add new unnamed connections
+          if (!foundConnection) {
+            let viewConn = particle.addUnnamedConnection();
```

Comment:
recipe slot is only created when it is provided, so if the particle consumes a slot, there only is the slot connection object at some point, until it is resolved (there is a particle that provides a slot for this connection).
but even before the consumed slot connection is resolved, it can be tied to one of the particle's view connections, so this is why it is in the slot connection. 
Am i not getting it right?